### ISSUE TITLE
Can't edit circles

### ISSUE THREAD

**Issue by Onixarts** [dceejay/RedMap#141]
Hi, I've defined some places on the map using circles on drawing layer. I then store them in the json file, load when map is activated - it works fine. But I can't edit these circles. When I click the circle with right mouse button the Add marker popup shows up, instead of shape edit mode as the manual says.

I'm sending circles using this code:

```js
if(map !== undefined){
    for(var i=0; i< map.markers.length; i++){
        let marker = {
            layer: 'markers',
            label: map.markers[i].name,
            editable: true,
            clickable: false 
        }
        Object.assign(marker, map.markers[i]) // here are some props like alt, lon and radius
        msg.payload.push(marker)
    }
}
```
It only happens when I click on circles added by code. When I draw new circle, I can edit it.
I don't know what am I doing wrong. Can sombody help me? :)

**Comment by dceejay** [comment]
Can you supply a working flow that shows the problem ? the code snippet above doesn't actually create a complete marker. I do note that you set the marker not clickable so when you click it is not responding and all you get is the add marker as if you clicked on the background.

**Comment by Onixarts** [comment]
Code above generates json array and it is just send to worldmap node. I have even copy that to inject node and the result is the same.

```json
[
    {
        "layer": "markers",
        "label": "astra",
        "editable": true,
        "clickable": true,
        "name": "astra",
        "lat": 51.126485353884796,
        "lon": 16.976081728935245,
        "radius": 93.18611875065358
    }
]
```
Enabling clickable just gives me abiliity to click and then label pops up. When clickable is false, then RMB cause new marker to be added. In the image below the left circle was created with json payload, the right is just clicked on the map. Then on both circles I have clicked RMB.

![image](https://pcqjaa.db.files.1drv.com/y4mfbmwg0g7zMpwM_hEdelvQyevFE0A3JgSnkMIVn70ZVE9Q49pdhCBkN5QBmmLHaQKN52sbT_qfaaPojaqAIh-aT1_xUw3L1KKehvQLAo-YqIBf2PtRdYb56uS-QiRiix_zIGLqxPaP81Z3FvtQCvHPs_wPM2Jwsb0596HAYGFNFZ2toWkkiSmjREaTRGA0HYV89Hfye20kSRqKho90y4F23PGFJGW135ITP_GiycI4PQ/worldmap-markers-problem.png?psid=1)

The circles html looks the same:

```html
<svg pointer-events="none" class="leaflet-zoom-animated" width="1658" height="602"
    style="transform: translate3d(-348px, -14px, 0px);" viewBox="-348 -14 1658 602">
    <g>
        <path class="leaflet-interactive" stroke="#910000" stroke-opacity="1" stroke-width="2" stroke-linecap="round"
            stroke-linejoin="round" fill="#910000" fill-opacity="0.2" fill-rule="evenodd"
            d="M-66.08648888766766,304.98323898017406a200,200 0 1,0 400,0 a200,200 0 1,0 -400,0 "></path>
        <path class="leaflet-interactive" stroke="#910000" stroke-opacity="0.5" stroke-width="4" stroke-linecap="round"
            stroke-linejoin="round" fill="#910000" fill-opacity="0.2" fill-rule="evenodd"
            d="M267.5,195.99910828471184a111,111 0 1,0 222,0 a111,111 0 1,0 -222,0 "></path>
    </g>
</svg>
```
So I guess there is a problem with JS click/mouseup event handler.

**Comment by dceejay** [comment]
Thanks - yes - for some "reason" I wasn't adding the edit to the rightmenu on injected shapes....
fixed in 2.4.2
